server/blob/blob-cloud/src/main/java/org/apache/james/blob/cloud/CloudsBlobsConfiguration.java

```diff
@@ -0,0 +1,115 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.cloud;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.jclouds.openstack.keystone.config.KeystoneProperties;
+import org.jclouds.openstack.swift.v1.reference.TempAuthHeaders;
+
+public class CloudsBlobsConfiguration {
+    private final URI endpoint;
+    private final Optional<Region> region;
+    private final Identity identity;
+    private final Properties overrides;
+    private final Credentials credentials;
+
+    public static class Builder {
```

Comment:
static things should come first (and checkstyle should be able to warn you)

Reply:
I opened https://issues.apache.org/jira/browse/JAMES-2535 as the scope is to large to be included in JAMES-2525